lib/cublas/linalg.jl

```diff
@@ -443,6 +443,68 @@ for (t, uploc, isunitc) in ((:LowerTriangular, 'U', 'N'),
     end
 end
 
+function LinearAlgebra.mul!(X::DenseCuMatrix{T},
+                            A::LowerTriangular{T,<:DenseCuMatrix},
+                            B::UpperTriangular{T,<:DenseCuMatrix},
+                            ) where {T<:CublasFloat}
+    triu!(parent(B))
+    trmm!('L', 'L', 'N', 'N', one(T), parent(A), parent(B), parent(X))
+    X
+end
+
+function LinearAlgebra.mul!(X::DenseCuMatrix{T},
+                            A::UpperTriangular{T,<:DenseCuMatrix},
+                            B::LowerTriangular{T,<:DenseCuMatrix},
+                            ) where {T<:CublasFloat}
+    tril!(parent(B))
+    trmm!('L', 'U', 'N', 'N', one(T), parent(A), parent(B), parent(X))
+    X
+end
+
+for (trtype, valtype) in ((:Transpose, :CublasFloat),
+                          (:Adjoint,   :CublasReal),
+                          (:Adjoint,   :CublasComplex))
+    @eval begin
+        function LinearAlgebra.mul!(X::DenseCuMatrix{T},
+                                    A::UpperTriangular{T,<:DenseCuMatrix},
+                                    B::LowerTriangular{<:Any,<:$trtype{T,<:DenseCuMatrix}},
+                                    ) where {T<:$valtype}
+            # operation is reversed to avoid executing the tranpose
+            triu!(parent(A))
+            CUBLAS.trmm!('R', 'U', 'T', 'N', one(T), parent(parent(B)), parent(A), parent(X))
+            X
+        end
+
+        function LinearAlgebra.mul!(X::DenseCuMatrix{T},
+                                    A::UpperTriangular{<:Any,<:$trtype{T,<:DenseCuMatrix}},
+                                    B::LowerTriangular{T,<:DenseCuMatrix},
+                                    ) where {T<:$valtype}
+            tril!(parent(B))
+            CUBLAS.trmm!('L', 'L', 'T', 'N', one(T), parent(parent(A)), parent(B), parent(X))
```

Comment:
Same here, and others below.

Reply:
@maleadt But the other existing routines use the BLAS prefix explicitly. For example:

```julia
@inline function LinearAlgebra.mul!(y::CuVector{T}, A::Hermitian{T,<:CuMatrix}, x::CuVector{T},
             α::Number, β::Number) where {T<:CublasComplex}
    alpha, beta = promote(α, β, zero(T))
    if alpha isa Union{Bool,T} && beta isa Union{Bool,T}
        return CUBLAS.hemv!(A.uplo, alpha, A.data, x, beta, y)
    else
        error("only supports BLAS type, got $T")
    end
end
```

in line 520. Is there a reason to use a different convention?